libs/ddd-angular/src/helpers/domain-helper.ts

```diff
@@ -1,19 +1,97 @@
 import { Tree, readProjectConfiguration } from '@nrwl/devkit';
+import { DDDObject } from '../model/ddd.model';
+import { Domain } from '../model/domain.model';
+import { API } from '../model/api.model';
+import { Shell } from '../model/shell.model';
+import { Feature } from '../model/feature.model';
+import { UI } from '../model/ui.model';
+import { Util } from '../model/util.model';
 
+/**
+ * It throws an error if the domain is invalid or shared
+ *
+ * @param tree The virtual filesystem tree provided by NX
+ * @param domain The domain name
+ */
 export const guardValidDomain = (tree: Tree, domain: string) => {
   if (!domainExist(tree, domain)) throw new Error('Invalid domain');
   if (domain === 'shared') throw new Error('Shared cannot be a domain');
 };
 
+/**
+ * It checks if the project is a domain by checking if it has the tag 'type:domain-logic'
+ * It throws an error if the domain is invalid
```

Comment:
It throws an error? :eyes:

Reply:
haha yeah the child throws an error, so raising it higher :D